### PR TITLE
Benchmark to test SingleRequestResponseToRemoteEntity with a local proxy

### PR DESCRIPTION
**Pull request by carlcamilleri**
Adding benchmark to test performance of sending messages to a remote entity without relying on remote `ASK` as per the approach discussed at https://github.com/akkadotnet/akka.net/issues/5203#issuecomment-907587322

### CODE REVIEW

File: src/benchmark/Akka.Cluster.Benchmarks/Sharding/ShardMessageRoutingBenchmarks.cs [hunk 1]
```diff
@@ -43,6 +45,50 @@ public class ShardMessageRoutingBenchmarks
         private IActorRef _batchActor;
         private Task _batchComplete;
 
+#if (DEBUG)
```
Comment (Aaronontheweb) [inline src/benchmark/Akka.Cluster.Benchmarks/Sharding/ShardMessageRoutingBenchmarks.cs:48]: What's this for?

Comment (carlcamilleri) [reply src/benchmark/Akka.Cluster.Benchmarks/Sharding/ShardMessageRoutingBenchmarks.cs:48]: I was using this to debug the functionality of the benchmark (be able to attach breakpoints etc.) as per https://benchmarkdotnet.org/articles/guides/troubleshooting.html - if another approach is usually taken, happy to remove or change as needed

Thanks

Comment (Aaronontheweb) [reply src/benchmark/Akka.Cluster.Benchmarks/Sharding/ShardMessageRoutingBenchmarks.cs:48]: Anything different about code in this block vs. the `#else` block other than `async Task` vs. `void`? Was trying to spot some but none jumped out at me. The other statement, with the debugging config added, makes sense to me.

Comment (carlcamilleri) [reply src/benchmark/Akka.Cluster.Benchmarks/Sharding/ShardMessageRoutingBenchmarks.cs:48]: Correct I was running into an issue with async code in debug mode - the benchmark was executing before the task completion. I can refactor for a cleaner approach if it’s better (still would need two methods I believe due to async signature)

Comment (Aaronontheweb) [reply src/benchmark/Akka.Cluster.Benchmarks/Sharding/ShardMessageRoutingBenchmarks.cs:48]: Got it. Thank you!

Nope, that's ok - just adding a comment to explain why it's there would be sufficient. We can do that in a subsequent PR.

Thank you for your contribution!

**Pull request merged**